x-pack/platform/packages/shared/kbn-es-snapshot-loader/src/replay/pipeline.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Client } from '@elastic/elasticsearch';
+import type { Logger } from '@kbn/logging';
+
+const TIMESTAMP_TRANSFORM_SCRIPT = `
+  if (ctx.containsKey('@timestamp') && ctx['@timestamp'] != null) {
+    Instant maxTime = Instant.parse(params.max_timestamp);
```

Comment:
If i understand this correctly, `maxTime` comes from the time the snapshot was created.  
So if I create a snapshot from data where the newest document is, say, a week old, wouldn't the replayed data still appear a week old instead of "now"?  Is the assumption we should always make sure to only create snapshots on "live" data?